fuse-ts-kdenlive.c

```diff
@@ -124,6 +124,28 @@ void close_kdenlive_project_file () {
 	}
 }
 
+static int parse_time_string(const char *in_str)
+{
+	double sum = 0;
+	const char *token_ptr = in_str;
+	const char *sep_ptr;
+	while ((sep_ptr = strchr(token_ptr, ':')))
+	{
+		double t = atof(token_ptr);
+		if (sum < 0)
+			sum = 60*sum - t;
+		else
+			sum = 60*sum + t;
+		token_ptr = sep_ptr+1;
+	}
+	double t = atof(token_ptr);
```

Comment:
there should be a NULL check between the loop and the "final" atof

Reply:
What kind of check? `token_prt` points within the string (at worst, the final null byte) so it will be a valid pointer. And in the case of the null byte `atof` will just return `0`, the same thing I would return in case this string is malformed / in unexpected / changed format.

Reply:
Or do mean we should return a fuse write error in this case?

Reply:
I was unsure what atof returns when presented with a pointer to a null byte. If it is safe to pass a pointer to 0 then it's fine.